### ISSUE TITLE
v9 `leave` within useTransition misses unit when interpolating from unit-less `0`

### ISSUE THREAD

**Issue by danielberndt** [react-spring/react-spring#779]
## 🐛 Bug Report

Here's my code:

```jsx
const OuterLayout = () => {
  const [show, setShow] = React.useState(false);
  const transitions = useTransition(show, null, {
    unique: true,
    from: {opacity: 0, transform: "translate3d(0,70%,0)"},
    initial: {opacity: 1, transform: "translate3d(0,0,0)"},
    enter: {opacity: 1, transform: "translate3d(0,0,0)"},
    leave: {opacity: 0, transform: "translate3d(0,50%,0)"},
  });
  return (
    <div>
      {transitions.map(
        ({item, key, props}) =>
          (item && console.log("0", props.transform.calc(0), "1", props.transform.calc(1))) ||
          (item && (
            <animated.div key={key} style={props} > + </animated.div>
          ))
      )}
    </div>
  );
};
```

the accompanying `console.log` looks like this

```
"0" "translate3d(0,0,0)" "1" "translate3d(0,0,0)"
"0" "translate3d(0,0,0)" "1" "translate3d(0,50,0)"
"0" "translate3d(0,0,0)" "1" "translate3d(0,50,0)"
```

as you can see, the `50` misses its unit. And thus it doesn't get animated. Note that it only happens on the first transition.

- `react-spring` 9.0.0-beta.31 (but also on latest v8)
- `react` v16.9.0

**Comment by danielberndt** [comment]
Ah, I just found out that this can be fixed by adding the corresponding unit to the `0` in `initial` i.e. `translate3d(0,0%,0)`.

**Comment by loeck** [comment]
I think for avoid this kind of issues juste use only number on from/enter/leave/initial values and use interpolation on render.

Example:

```
const OuterLayout = () => {
  const [show, setShow] = React.useState(false);
  const transitions = useTransition(show, null, {
    unique: true,
    from: {opacity: 0, transform: 70},
    initial: {opacity: 1, transform: 0},
    enter: {opacity: 1, transform: 0},
    leave: {opacity: 0, transform: 50},
  });
  return (
    <div>
      {transitions.map(
        ({item, key, props}) =>
          (item && console.log("0", props.transform.calc(0), "1", props.transform.calc(1))) ||
          (item && (
            <animated.div key={key} style={{
              opacity: props.opacity,
              transform: props.transform.interpolate(v => `translate3d(0,${v}%,0)`)
            }} > + </animated.div>
          ))
      )}
    </div>
  );
};
```

**Comment by danielberndt** [comment]
Thanks, it's good to know that there are work arounds, but I still feel like this is a bug that might affect and confuse quite some people, especially since adding a unit to a transition starting from `0` seems to works with `enter` but not with `initial`.

**Comment by aleclarson** [comment]
> especially since adding a unit to a transition starting from `0` seems to works with `enter` but not with `initial`.

Please provide a sandbox that demonstrates this claim. Thanks!

**Comment by danielberndt** [comment]
repro is here: https://codesandbox.io/s/elegant-platform-1xu4s

check out the console.logs and note that the `transform` won't be correctly applied when clicking the button for the first time.

**Comment by aleclarson** [comment]
Fixed in #809

You can try it today: `npm install react-spring@9.0.0-canary.809.3.a66b280`

Here's your example with the required changes:

```js
import React from 'react'
import { useTransition, a } from 'react-spring'

const Demo = () => {
  const [state, set] = React.useState(1)
  const transition = useTransition(state, {
    from: { opacity: 0, transform: 'translate3d(0,70%,0)' },
    initial: { opacity: 1, transform: 'translate3d(0,0%,0)' },
    enter: { opacity: 1, transform: 'translate3d(0,0%,0)' },
    leave: { opacity: 0, transform: 'translate3d(0,50%,0)' }
  })
  return (
    <div>
      <div>
        <button onClick={() => set(state => state + 1)}>click</button>
      </div>
      <div>
        {transition((props, item) => (
		  item % 2 > 0 && <a.div style={{ position: 'absolute', ...props }}>+</a.div>
		))}
      </div>
    </div>
  )
}

export default Demo
```
